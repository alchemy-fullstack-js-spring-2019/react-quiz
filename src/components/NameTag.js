//container component
//contains state
//uses the other components
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import ColorSelector from './ColorSelector';
import TextSelector from './TextSelector';
import Shape from './Shape';

export default class NameTag extends PureComponent {
  //this is where I get confused with modularization. How do I grab the text that's written in the text selector and store it in this state in order to pass it to shape? 
  //can do if all in the same component but getting confused when separating them
  state = {
    text: '',
    color: '',
    backgroundColor: ''
  }

  handleChange = ({ target }) => {
    this.setState({
      [target.name]: target.value
    });
  }

  render() {
    return (
      <>
        <TextSelector />
        <ColorSelector />
        <Shape color={this.state.color} backgroundColor={this.state.backgroundColor} text={this.state.text} />
      </>
    );
  }
}

