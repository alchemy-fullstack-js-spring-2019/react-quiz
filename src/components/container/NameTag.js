import React, { PureComponent } from 'react';
import TextSelector from '../presentational/TextSelector';
import ColorSelector from '../presentational/ColorSelector';
import Shape from '../presentational/Shape';

export default class NameTag extends PureComponent {
  state = {
    text: '',
    fontColor: '',
    backgroundColor: ''
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }


  render() {

    const { text, fontColor, backgroundColor } = this.state;
    return (
      <>
      <TextSelector handleChange={this.handleChange} />
      <ColorSelector color={backgroundColor} handleChange={this.handleChange} />
      <ColorSelector color={text} handleChange={this.handleChange} />
      <Shape text={text} backgroundColor={backgroundColor} fontColor={fontColor}/>
      </>
    );
  }
}
