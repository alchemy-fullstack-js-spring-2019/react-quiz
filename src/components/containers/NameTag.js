import React, { PureComponent } from 'react';
import TextSelector from '../TextSelector';
import ColorSelector from '../ColorSelector';
import Shape from '../Shape';

export default class NameTag extends PureComponent {
  render() {
    const { text, color, backgroundColor } = this.state;

    return (
      <>
        <TextSelector text={text} />
        <ColorSelector color={color} backgroundColor={backgroundColor} />
        <Shape />
      </>
    );
  }
}
