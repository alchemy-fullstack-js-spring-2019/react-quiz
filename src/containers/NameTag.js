import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import ColorSelector from '../components/ColorSelector';
import TextSelector from '../components/TextSelector';
import Shape from '../components/Shape';

export default class NameTag extends PureComponent {

  state = {
    text: 'hello',
    color: 'red',
    backgroundColor: 'black'
  }

  addText = text => {
    this.setState({ text });
  }

  render() {
    const { text, color, backgroundColor } = this.state;
    return (
      <>
        <TextSelector addText={this.addText}/>
        {/* <ColorSelector addColors={this.addColors}/> */}
        <Shape text={text} color={color} backgroundColor={backgroundColor} />
      </>
    );
  }
}

