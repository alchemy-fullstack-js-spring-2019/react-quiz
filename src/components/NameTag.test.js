import React from 'react';
import { shallow } from 'enzyme';
import NameTag from './NameTag';

describe('TextSelector component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<NameTag />);
    expect(wrapper).toMatchSnapshot();
  });
});
