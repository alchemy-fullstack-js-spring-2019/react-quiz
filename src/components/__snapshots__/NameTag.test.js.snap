// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component matches snapshot 1`] = `
<Fragment>
  <TextSelector
    change={[Function]}
    text=""
  />
  <ColorSelector
    backgroundColor=""
    change={[Function]}
    color=""
  />
  <hr />
  <Shape
    backgroundColor=""
    color=""
    text=""
  />
</Fragment>
`;
